### PR TITLE
Add documentation of the mqtt spec

### PR DESCRIPTION
**Pull request by vetlek**

### CODE REVIEW

File: mqtt_spec.yaml [hunk 1]
```diff
@@ -0,0 +1,287 @@
+asyncapi: 2.3.0
+info:
+  title: Flotilla MQTT API
+  version: 1.0.0
+  description: >-
+    Flotilla MQTT API is a streaming broker for producers and conumers of real
+    time data about robots and missions on an Equinor asset.
```
Comment (aestene) [inline mqtt_spec.yaml:7]:
```suggestion
    Flotilla MQTT API is a streaming broker for producers and cosnumers of real
    time data about robots and missions on facility.
```

File: mqtt_spec.yaml [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0
```
Comment (aestene) [inline mqtt_spec.yaml:10]: Shouldn't this be the license of Flotilla which is EPL 2.0?

File: mqtt_spec.yaml [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+    url: localhost:{port}
+    protocol: mqtt
+    description: Development broker
+    variables:
+      port:
+        default: '1883'
+        enum:
+          - '1883'
+    security:
+      - user-password: []
```
Comment (aestene) [inline mqtt_spec.yaml:22]: Can you add `mosquitto` as user and `default` as password here? Those as the default values that are openly available anyway.

Comment (vetlek) [reply mqtt_spec.yaml:22]: Couldn't find anything in the documentation about adding this, so don't think it is possible

File: mqtt_spec.yaml [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+      minimum: 0
+      maximum: 100
+    telemetryPressure:
+      type: number
+      format: float
+      description: The atex overpressure messured in mBar
+    sentAt:
+      type: string
+      format: date-time
+      description: Date and time when the message was sent.
```
Comment (aestene) [inline mqtt_spec.yaml:274]:
```suggestion
    timestamp:
      type: string
      format: date-time
      description: Date and time when the message was sent.
```
```suggestion
    sentAt:
      type: string
      format: date-time
      description: Date and time when the message was sent.
```

File: mqtt_spec.yaml [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+      payload:
+        $ref: '#/components/schemas/telemetryPressurePayload'
+  schemas:
+    isarStatePayload:
+      type: object
+      properties:
+        state:
+          $ref: '#/components/schemas/isarState'
+        sent_at:
+          $ref: '#/components/schemas/sentAt'
```
Comment (aestene) [inline mqtt_spec.yaml:168]:
```suggestion
        timestamp:
          $ref: '#/components/schemas/timestamp'
```
```suggestion
        sent_at:
          $ref: '#/components/schemas/sentAt'
```

File: mqtt_spec.yaml [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+      type: object
+      properties:
+        id:
+          type: string
+        tasks:
+          type: array
+          items:
+            $ref: '#/components/schemas/isarTask'
+        sent_at:
+          $ref: '#/components/schemas/sentAt'
```
Comment (aestene) [inline mqtt_spec.yaml:179]:
```suggestion
        timestamp:
          $ref: '#/components/schemas/timestamp'
```
```suggestion
        sent_at:
          $ref: '#/components/schemas/sentAt'
```

File: mqtt_spec.yaml [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+    telemetryPosePayload:
+      type: object
+      properties:
+        pose:
+          $ref: '#/components/schemas/telemetryPose'
+        sent_at:
+          $ref: '#/components/schemas/sentAt'
```
Comment (aestene) [inline mqtt_spec.yaml:186]:
```suggestion
        timestamp:
          $ref: '#/components/schemas/timestamp'
```

File: mqtt_spec.yaml [hunk 8]
```diff
@@ -0,0 +1,287 @@
...
+    telemetryBatteryPayload:
+      properties:
+        battery:
+          $ref: '#/components/schemas/telemetryBattery'
+        sent_at:
+          $ref: '#/components/schemas/sentAt'
```
Comment (aestene) [inline mqtt_spec.yaml:192]:
```suggestion
        timestamp:
          $ref: '#/components/schemas/timestamp'
```

**Review by aestene — CHANGES_REQUESTED** [review]
Have a look at the comments but otherwise this looks very good! I think it will be very useful :tada:

**Pull request merged**